src/main/java/com/upserve/uppend/AppendOnlyStore.java

```diff
@@ -1,13 +1,51 @@
 package com.upserve.uppend;
 
 import java.io.Flushable;
+import java.util.stream.Stream;
 
 /**
  * Add byte arrays under a key and partition, and retrieve them. Note the
  * expectation that the byte arrays are appended to the value, which is an
  * ever-growing list of byte arrays.
  */
 public interface AppendOnlyStore extends ReadOnlyAppendOnlyStore, Flushable {
+    /**
+     * Read byte arrays that have been stored under a given partition and key in
+     * parallel, skipping the write cache of this store so that unflushed data
+     * is not visible
+     *
+     * @param partition the partition under which to retrieve
+     * @param key the key under which to retrieve
+     * @throws IllegalArgumentException if partition is invalid
+     * @return a parallel stream of the stored byte arrays
+     */
+    Stream<byte[]> readFlushed(String partition, String key);
+
+    /**
+     * Read byte arrays that have been stored under a given partition and key in
+     * the order they were stored, skipping the write cache of this store so
+     * that unflushed data is not visible
+     *
+     * @param partition the partition under which to retrieve
+     * @param key the key under which to retrieve
+     * @throws IllegalArgumentException if partition is invalid
+     * @return a stream of the stored byte arrays in storage order
+     */
+    Stream<byte[]> readSequentialFlushed(String partition, String key);
```

Comment:
In what case would we not want to read in parallel?

Reply:
So you can read in order

Reply:
Understood, but what's the use case? When do we want to read in order?